mate-panel/panel.c

```diff
@@ -413,17 +413,36 @@ panel_key_press_event (GtkWidget   *widget,
 	return FALSE;
 }
 
+static gboolean
+can_set_background_image (PanelToplevel *toplevel,
+                          const gchar   *uri,
+                          gchar        **filename)
+{
+	gchar *image;
+
+	if (! panel_profile_background_key_is_writable (toplevel, "type") ||
+	    ! panel_profile_background_key_is_writable (toplevel, "image"))
+		return FALSE;
+
+	image = g_filename_from_uri (uri, NULL, NULL);
+	if (! image)
+		return FALSE;
+
+	if (filename)
+		*filename = image;
+	else
+		g_free (image);
```

Comment:
You don’t really need this if you use `*filename` directly.

Reply:
What do you mean?  I have one caller of this function that pass `NULL` as `filename` because it doesn't care what the value is, just whether the other checks succeeded.  Admittedly the ability to return the filename is a mere convenience and optimization not to re-compute the filename, but it seemed handy and fairly innocent to me as it's internal API anyway.

Reply:
Where did you set the filename's value in can_set_background_image?

Reply:
I don't, it's only an out variable.  But I use the value I may assign to that out variable to determine the return value of `can_set_background_image()`, regardless of whether that out variable is `NULL` or not.